dask_expr/datasets.py

```diff
@@ -51,7 +51,7 @@ def random_state(self):
             k: (
                 np.random.randint(2e9, size=npartitions)
                 if self.seed is None
-                else random_state_data(npartitions, self.seed)
+                else [self.seed] * npartitions
```

Comment:
Does this mean that every partition gets the same seed and might generate the same data?

Reply:
Oh, I missunderstood something, sorry. Thx!

Updated to avoid serializing all of them but keep the behaviour..